File: PRATICA_2/quadrantes.c
/*

* Program: [Quadrantes]

* Author: [Matheus H Medeiros]

* Date: [12/03/2024]

* Description: [Divide a tela em quato quadrantes e impime em qual desses quadrantes está o cursor]

* License: [CC BY]

*/

#include <stdio.h>
#include <windows.h>

int main() {
    POINT cursorPos;
    
    // Definições das variáveis da tela, de resolução 1920x1080p
    int altura_tela = 1080, comprimento_tela = 1920, quadrante_altura = 540, quadrante_comprimento = 960;
    int quit = 0;

    while (!quit) {
        GetCursorPos(&cursorPos);
		
		// Calculo que retorna 1,2,3 ou 0 para o switch case
        switch (cursorPos.x < quadrante_comprimento ? (cursorPos.y < quadrante_altura ? 0 : 2) : (cursorPos.y < quadrante_altura ? 1 : 3)) {
            case 0:
                printf("Quadrante A\n");
                break;
            case 1:
                printf("Quadrante B\n");
                break;
            case 2:
                printf("Quadrante C\n");
                break;
            case 3:
                printf("Quadrante D\n");
                break;
        }

        if (GetAsyncKeyState(VK_ESCAPE) & 0x8000) // Verifica se a tecla ESC está pressionada
            quit = 1;

        Sleep(200);
        system("cls");
    }

    return 0;
}

File: PRATICA_2/quadrantes.m
function quadrantes()
    % Definições das variáveis da tela, de resolução 1920x1080p
    quadrante_altura = 540;
    quadrante_comprimento = 960;
    quit = false;

    while ~quit
        cursorPos = get(0, 'PointerLocation');
		
		% Calculo que retorna 1,2,3 ou 0 para o switch case
        switch (cursorPos(1) < quadrante_comprimento) + 2 * (cursorPos(2) >= quadrante_altura)
            case 0
                disp('Quadrante D');    
            case 1
                disp('Quadrante C');
            case 2
                disp('Quadrante B');
            case 3
                disp('Quadrante A');
        end

        figure (1)
        
        tic
        while toc<0.01
            
            ch = get(gcf, 'CurrentCharacter');
        end
        
        if (ch == 'c')
            break

        end
    end
end
